static/app/components/notificationActions/notificationActionManager.tsx

```diff
@@ -0,0 +1,203 @@
+import {Fragment, useState} from 'react';
+import capitalize from 'lodash/capitalize';
+
+import DropdownButton from 'sentry/components/dropdownButton';
+import {DropdownMenu, MenuItemProps} from 'sentry/components/dropdownMenu';
+import NotificationActionItem from 'sentry/components/notificationActions/notificationActionItem';
+import {IconAdd} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {Project} from 'sentry/types';
+import {
+  AvailableNotificationAction,
+  NotificationAction,
+  NotificationActionService,
+} from 'sentry/types/notificationActions';
+
+type NotificationActionManagerProps = {
+  /**
+   * The list of existing notification actions
+   */
+  actions: NotificationAction[];
+  /**
+   * The list of available notification actions
+   */
+  availableActions: AvailableNotificationAction[];
+  /**
+   * The project associated with the notification actions
+   * TODO(enterprise): refactor to account for multiple projects
+   */
+  project: Project;
+  /**
+   * Updates the notification alert count for this project
+   */
+  updateAlertCount: (projectId: number, alertCount: number) => void;
+  /**
+   * Optional list of roles to display as recipients of Sentry notifications
+   */
+  recipientRoles?: string[];
+};
+
+const NotificationActionManager = ({
+  actions,
+  availableActions,
+  recipientRoles,
+  project,
+  updateAlertCount,
+}: NotificationActionManagerProps) => {
+  const [notificationActions, setNotificationActions] =
+    useState<Partial<NotificationAction>[]>(actions);
+
+  const addNotificationAction = (
+    actionInstance: AvailableNotificationAction['action']
+  ) => {
+    const updatedActions = [...notificationActions, actionInstance];
+    setNotificationActions(updatedActions);
+    updateAlertCount(parseInt(project.id, 10), updatedActions.length);
+  };
+
+  const removeNotificationAction = (index: number) => {
```

Comment:
so both the add and remove methods here are modifying state locally - which is ok

Ideally, we'd actually fetch actions from teh server again so we can be sure our client & server state are always in sync.

Reply:
the adding and removing are used to only modify state since when you add a `NotificationActionItem` and don't save, it should still show up in state (but won't be present in the db). similarly for when you have an unsaved `NotificationActionItem` that's not in the db and you press "Cancel" -- it should remove it from state

PUT / POST / DELETE happens in the `NotificationActionItem` component and we also call these functions to update state if the requests are successful

Reply:
Ahhh i see i see 👍

Reply:
I wonder if there's an better way to name the add/remove methods then 🤔 
to make it more clear that this is a 'staged' change that hasn't been submitted yet....